### PR TITLE
Update CI build to run tests

### PR DESCRIPTION
**Pull request by jasonleenaylor**

### CODE REVIEW

File: .github/workflows/fw_icu4c_ci.yml [hunk 1]
```diff
@@ -32,6 +31,38 @@ jobs:
...
+    - name: Prepare data for SIL tests (64 bit - TODO need this?)
+      working-directory: icu4c
+      run: msbuild nugetpackage/build/icu4c.proj /t:PrepareTestData
+
+    - name: Package (both 32bit and 64bit architectures)
+      working-directory: icu4c
+      run: msbuild nugetpackage/build/icu4c.proj /t:BuildPackage
+
+    - name: Upload Artifacts
+      uses: actions/upload-artifact@v2
```
Comment (ermshiperete) [inline .github/workflows/fw_icu4c_ci.yml:56]: Where do these get uploaded to?

File: .github/workflows/fw_icu4c_ci.yml [hunk 2]
```diff
@@ -32,6 +31,38 @@ jobs:
...
+      run: msbuild source/allinone/allinone.sln /p:Configuration=Release /p:Platform=Win32 /p:SkipUWP=true
+      if: ${{ matrix.os == 'windows-latest' }}
+
+    - name: Prepare data for SIL tests
+      working-directory: icu4c
+      run: msbuild nugetpackage/build/icu4c.proj /t:PrepareTestData
+
+    - name: Test ICU 32 bit
+      run: source/allinone/icucheck.bat x86 Release
+      working-directory: icu4c
```
Comment (ermshiperete) [inline .github/workflows/fw_icu4c_ci.yml:41]: The GH action fails in this step. Although, looking at the build log it's not very clear to my why. The tests seem to pass and yet the end result is failed? 

Ah, I see. The `iotest` test suite failed.

**Review by ermshiperete — APPROVED** [review]

File: .github/workflows/fw_icu4c_ci.yml (hunk 2 above)
Comment (jasonleenaylor) [reply .github/workflows/fw_icu4c_ci.yml:41]: Yes, when I changed the runtime that test suite started failing. :-(

File: .github/workflows/fw_icu4c_ci.yml (hunk 1 above)
Comment (jasonleenaylor) [reply .github/workflows/fw_icu4c_ci.yml:56]: They are added to the artifacts section of the github action results.

**Review by papeh — APPROVED** [review]
:lgtm:

> Reviewed 1 of 2 files at r1, 66 of 66 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/sillsdev/icu/16#-:-MwDtu3G311n00Ka-DVk:bnfp4nl)* status: :shipit: complete! all files reviewed, all discussions resolved (waiting on @jasonleenaylor)

**Pull request merged**